SmartDeviceLink/SDLResponseDispatcher.m

```diff
@@ -140,6 +143,43 @@ - (void)clear {
     _audioPassThruHandler = nil;
 }
 
+- (void)sdl_removeOldButtonHandlersFor:(SDLRPCRequest *)request {
+    for(SDLRPCCorrelationId *key in self.rpcRequestDictionary.allKeys) {
+        if ([self.rpcRequestDictionary[key] isKindOfClass:[SDLAlert class]] && [request isKindOfClass:[SDLAlert class]]) {
```

Comment:
We already know the request class type based on where this came from, ideally, we don't want to re-check it possibly hundreds of times in this loop. Instead of one big method, there should be smaller `sdl_removeButtonHandlersFor:(SDLAlert *)alert` (and such for the others') as well.